engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -842,6 +878,707 @@ static int SetTexture(lua_State* L)
 }
 
+// Allocates a new array and fills it with data from a lua table at top of stack.
+// Only supports number values. Note: Doesn't do any error checking!
+template<typename T>
+static void MakeNumberArrayFromLuaTable(lua_State* L, const char* field, void** data_out, uint32_t* count_out)
+{
+    lua_getfield(L, -1, field);
+    int num_entries = lua_objlen(L, -1);
+    T* data_ptr = new T[num_entries];
+
+    lua_pushnil(L);
+    while (lua_next(L, -2))
+    {
+        int32_t table_index     = lua_tonumber(L, -2);
+        data_ptr[table_index-1] = lua_tonumber(L, -1);
+        lua_pop(L, 1);
+    }
+    lua_pop(L, 1);
+    *data_out  = data_ptr;
+    *count_out = num_entries;
+}
+
+static void DestroyTextureSet(dmGameSystemDDF::TextureSet& texture_set)
+{
+    delete[] texture_set.m_Animations.m_Data;
+    delete[] texture_set.m_Geometries.m_Data;
+    delete[] texture_set.m_FrameIndices.m_Data;
+}
+
+// These lookup functions are needed because the values for the two enums are different,
+// so we can't rely on the raw value to convert between them
+static dmGameObject::Playback DDFPlaybackToGameObjectPlayback(dmGameSystemDDF::Playback playback)
+{
+    switch(playback)
+    {
+        case dmGameSystemDDF::PLAYBACK_NONE          : return dmGameObject::PLAYBACK_NONE;
+        case dmGameSystemDDF::PLAYBACK_ONCE_FORWARD  : return dmGameObject::PLAYBACK_ONCE_FORWARD;
+        case dmGameSystemDDF::PLAYBACK_ONCE_BACKWARD : return dmGameObject::PLAYBACK_ONCE_BACKWARD;
+        case dmGameSystemDDF::PLAYBACK_ONCE_PINGPONG : return dmGameObject::PLAYBACK_ONCE_PINGPONG;
+        case dmGameSystemDDF::PLAYBACK_LOOP_FORWARD  : return dmGameObject::PLAYBACK_LOOP_FORWARD;
+        case dmGameSystemDDF::PLAYBACK_LOOP_BACKWARD : return dmGameObject::PLAYBACK_LOOP_BACKWARD;
+        case dmGameSystemDDF::PLAYBACK_LOOP_PINGPONG : return dmGameObject::PLAYBACK_LOOP_PINGPONG;
+        default:break;
+    }
+    assert(0);
+    return (dmGameObject::Playback) -1;
+}
+
+static dmGameSystemDDF::Playback GameObjectPlaybackToDDFPlayback(dmGameObject::Playback playback)
+{
+    switch(playback)
+    {
+        case dmGameObject::PLAYBACK_NONE:          return dmGameSystemDDF::PLAYBACK_NONE;
+        case dmGameObject::PLAYBACK_ONCE_FORWARD:  return dmGameSystemDDF::PLAYBACK_ONCE_FORWARD;
+        case dmGameObject::PLAYBACK_ONCE_BACKWARD: return dmGameSystemDDF::PLAYBACK_ONCE_BACKWARD;
+        case dmGameObject::PLAYBACK_ONCE_PINGPONG: return dmGameSystemDDF::PLAYBACK_ONCE_PINGPONG;
+        case dmGameObject::PLAYBACK_LOOP_FORWARD:  return dmGameSystemDDF::PLAYBACK_LOOP_FORWARD;
+        case dmGameObject::PLAYBACK_LOOP_BACKWARD: return dmGameSystemDDF::PLAYBACK_LOOP_BACKWARD;
+        case dmGameObject::PLAYBACK_LOOP_PINGPONG: return dmGameSystemDDF::PLAYBACK_LOOP_PINGPONG;
+        default:break;
+    }
+    assert(0);
+    return (dmGameSystemDDF::Playback) -1;
+}
+
+static void ValidateSetAtlasAnimations(lua_State* L, uint32_t* num_animations_out)
```

Comment:
If this function received the number of geometries, it would be easy to check if the frames are out of range?